integration/ksm/ksm_test.sh

```diff
@@ -0,0 +1,88 @@
+#!/bin/bash
+#
+# Copyright (c) 2019 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+set -e
+
+dir_path=$(dirname "$0")
+source "${dir_path}/../../lib/common.bash"
+source "${dir_path}/../../metrics/lib/common.bash"
+
+TEST_NAME="Run containers with KSM"
+KATA_KSM_THROTTLER="${KATA_KSM_THROTTLER:-yes}"
+RUNTIME="${RUNTIME:-kata-runtime}"
+PAYLOAD_ARGS="${PAYLOAD_ARGS:-tail -f /dev/null}"
+IMAGE="${IMAGE:-busybox}"
+WAIT_TIME="60"
+
+function turn_on_ksm() {
+	echo "Turning on KSM"
+	echo "1" | sudo tee "${KSM_ENABLE_FILE}"
+}
+
+function setup() {
+	clean_env
+
+	check_processes
+
+	turn_on_ksm
+
+	check_for_ksm
+}
+
+function teardown() {
+	clean_env
+
+	check_processes
+
+	save_ksm_settings
```

Comment:
Saving ksm settings right before restoring them? Shouldn't they be saved on the `setup` before `turn_on_ksm`?

Reply:
@chavafg the reason why I am putting the `save_ksm_settings` there is because we need to run it first to use `restore_ksm`
https://github.com/kata-containers/tests/blob/10bdf7374f7c31263d788404496e9b8c32a1999a/metrics/lib/common.bash#L250